hw/ip/flash_ctrl/data/flash_ctrl.hjson.tpl

```diff
@@ -784,7 +798,94 @@
         { bits: "2",    name: "prog_full",  desc: "Flash program FIFO full"},
         { bits: "3",    name: "prog_empty", desc: "Flash program FIFO empty, software must provide data", resval: "1"},
         { bits: "4",    name: "init_wip",   desc: "Flash controller undergoing init, inclusive of phy init"},
-        { bits: "${8 + bank_width + page_width -1}:8", name: "error_addr", desc: "Flash controller error address."},
+      ]
+    },
+
+    { name: "ERR_CODE",
+      desc: '''
+        Flash error code register.
+        This register tabulates detailed error status of the flash.
+        This is separate from !!OP_STATUS, which is used to indicate the current state of the software initiated
+        flash operation.
+      '''
+      swaccess: "rw",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "0",
+          name: "flash_err",
+          desc: '''
+            The flash memory itself has an error, please check the vendor specs for details of the error.
+          '''
+        },
+        { bits: "1",
+          name: "flash_alert",
+          desc: '''
+            The flash memory itself has triggered an alert, please check the vendor specs for details of the error.
+          '''
+        },
+        { bits: "2",
+          name: "mp_err",
+          desc: '''
+            Flash access has encountered an access permission error.
+            Please see !!ERR_ADDR for exact address.
+          '''
+        },
+        { bits: "3",
+          name: "ecc_single_err",
+          desc: '''
+            Flash access has encountered a recoverable ECC error.
+            Please see !!ECC_ERR_ADDR for exact address.
+          '''
+        },
+        { bits: "4",
+          name: "ecc_multi_err",
+          desc: '''
+            Flash access has encountered a non-recoverable ECC error.
+            Please see !!ECC_ERR_ADDR for exact address.
+          '''
+        },
+      ]
+    },
+
+    { name: "ERR_ADDR",
+      desc: "Access permission error address",
+      swaccess: "ro",
+      hwaccess: "hwo",
+      fields: [
+        { bits: "${bank_width + page_width -1}:0",
+          resval: 0,
+        },
+      ]
+    },
+
+    { multireg: {
+        cname: "ECC_ERR"
+        name: "ECC_ERR_ADDR",
+        desc: "ecc error address",
+        count: "RegNumBanks",
+        swaccess: "ro",
+        hwaccess: "hwo",
+        fields: [
+          { bits: "${total_byte_width-1}:0",
+            resval: 0,
+          },
+        ]
+      }
+    },
+
+    { name: "PHY_ALERT_CFG",
+      desc: "Phy alert configuration",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          name: "alert_ack",
+          desc: "Acknowledge flash phy alert"
+        },
+        { bits: "1",
+          name: "alert_trig",
+          desc: "Trigger flash phy alert"
+        }
```

Comment:
given that this alert condition can be cleared by SW without a system reset, would it make sense to make the corresponding alert recoverable instead of fatal?

Reply:
yep you're right, changing to recoverable